### ISSUE TITLE
ROCm docker container fails to build

### ISSUE THREAD

**Issue by Kautenja** [microsoft/onnxruntime#16445]
### Describe the issue

The ROCm docker container fails to build using the provided install commands.

### Target platform

ROCm

### Build script

https://github.com/microsoft/onnxruntime/blob/main/dockerfiles/README.md#rocm

### Error / output

```
#0 921.4 1 warning generated when compiling for gfx90a.
#0 1228.6 [ 82%] Linking CXX shared module libonnxruntime_providers_rocm.so
#0 1232.3 /usr/bin/ld: CMakeFiles/onnxruntime_providers_rocm.dir/amdgpu/onnxruntime/core/providers/rocm/rocm_provider_interface.cc.o: in function `GetProvider':
#0 1232.3 rocm_provider_interface.cc:(.text.GetProvider+0x0): multiple definition of `GetProvider'; CMakeFiles/onnxruntime_providers_rocm.dir/code/onnxruntime/onnxruntime/core/providers/rocm/rocm_provider_factory.cc.o:rocm_provider_factory.cc:(.text.GetProvider+0x0): first defined here
#0 1232.3 collect2: error: ld returned 1 exit status
#0 1232.3 make[2]: *** [CMakeFiles/onnxruntime_providers_rocm.dir/build.make:3390: libonnxruntime_providers_rocm.so] Error 1
#0 1232.3 make[1]: *** [CMakeFiles/Makefile2:3071: CMakeFiles/onnxruntime_providers_rocm.dir/all] Error 2
#0 1232.3 make: *** [Makefile:166: all] Error 2
#0 1232.3 rocm_home = /opt/rocm
#0 1232.3 Traceback (most recent call last):
#0 1232.3   File "/code/onnxruntime/tools/ci_build/build.py", line 2640, in <module>
#0 1232.3     sys.exit(main())
#0 1232.3   File "/code/onnxruntime/tools/ci_build/build.py", line 2533, in main
#0 1232.3     build_targets(args, cmake_path, build_dir, configs, num_parallel_jobs, args.target)
#0 1232.3   File "/code/onnxruntime/tools/ci_build/build.py", line 1455, in build_targets
#0 1232.3     run_subprocess(cmd_args, env=env)
#0 1232.3   File "/code/onnxruntime/tools/ci_build/build.py", line 792, in run_subprocess
#0 1232.3     return run(*args, cwd=cwd, capture_stdout=capture_stdout, shell=shell, env=my_env)
#0 1232.3   File "/code/onnxruntime/tools/python/util/run.py", line 57, in run
#0 1232.3     shell=shell,
#0 1232.3   File "/opt/miniconda/lib/python3.7/subprocess.py", line 468, in run
#0 1232.3     output=stdout, stderr=stderr)
#0 1232.3 subprocess.CalledProcessError: Command '['/code/cmake-3.26.3-linux-x86_64/bin/cmake', '--build', '/code/onnxruntime/build/Linux/Release', '--config', 'Release', '--', '-j192']' returned non-zero exit status 2.
------
Dockerfile.rocm:18
--------------------
  17 |     # Prepare onnxruntime repository & build onnxruntime
  18 | >>> RUN git clone --single-branch --branch ${ONNXRUNTIME_BRANCH} --recursive ${ONNXRUNTIME_REPO} onnxruntime &&\
  19 | >>>     /bin/sh onnxruntime/dockerfiles/scripts/install_common_deps.sh &&\
  20 | >>>     cd onnxruntime &&\
  21 | >>>     /bin/sh ./build.sh --allow_running_as_root --config Release --build_wheel --update --build --parallel --cmake_extra_defines\
  22 | >>>             ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) --use_rocm --rocm_home=/opt/rocm &&\
  23 | >>>     pip install /code/onnxruntime/build/Linux/Release/dist/*.whl &&\
  24 | >>>     cd ..
  25 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c git clone --single-branch --branch ${ONNXRUNTIME_BRANCH} --recursive ${ONNXRUNTIME_REPO} onnxruntime &&    /bin/sh onnxruntime/dockerfiles/scripts/install_common_deps.sh &&    cd onnxruntime &&    /bin/sh ./build.sh --allow_running_as_root --config Release --build_wheel --update --build --parallel --cmake_extra_defines            ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) --use_rocm --rocm_home=/opt/rocm &&    pip install /code/onnxruntime/build/Linux/Release/dist/*.whl &&    cd .." did not complete successfully: exit code: 1
```

### Visual Studio Version

NA

### GCC / Compiler Version

NA

**Comment by mindest** [comment]
Fixed in #16458. Please try another build, and make sure the main branch is updated when building the docker.

**Comment by Kautenja** [comment]
This indeed resolves my issue. Thank you!
